### PR TITLE
GitHub Releases BullMQ Automation

### PR DESCRIPTION
**Pull request by Ferotiq**
### Description

Adds functionality to send new GitHub releases (triggered from GitHub Webhook) to George.

Closes #

### CODE REVIEW

File: backend/src/api/routes/webhooks.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+const router = Router();
+
+router.post(
+  "/releases",
+  RateLimit.webhooksPost,
+  validateRequest({
+    body: {
+      action: joi.string().equal("published").required(),
+      release: joi
```
Comment (Ferotiq) [inline backend/src/api/routes/webhooks.ts:15]: Adding the entire schema is probably best practice here but I didn't do that because it would take a while

**Comment by Ferotiq** [comment]
I still need to verify that the request is coming from GitHub and not some rando

File: backend/src/api/controllers/webhook.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import crypto from "crypto";
+import georgeQueue from "../../queues/george-queue";
+import { MonkeyResponse } from "../../utils/monkey-response";
+
+const secretHash = process.env.GITHUB_WEBHOOK_SECRET
+  ? crypto
+      .createHash("sha256")
+      .update(process.env.GITHUB_WEBHOOK_SECRET)
+      .digest("hex")
+  : null;
```
Comment (Miodec) [inline backend/src/api/controllers/webhook.ts:10]: I don't think it makes sense to generate a new secret when comparing to the one that's being sent from GitHub.

File: backend/src/api/controllers/webhook.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+export async function sendRelease(
+  req: MonkeyTypes.Request
+): Promise<MonkeyResponse> {
+  const { "x-hub-signature-256": signature } = req.headers;
+  if (secretHash && signature !== `sha256=${secretHash}`) {
+    return new MonkeyResponse("Unauthorized", 401);
+  }
+
```
Comment (Miodec) [inline backend/src/api/controllers/webhook.ts:19]: Add a 500 here if the .env property is not set

File: backend/src/api/routes/webhooks.ts (hunk 1 above)
Comment (Miodec) [reply backend/src/api/routes/webhooks.ts:15]: Thats fair, but i think joi will fail the validation if there are extra properties compared to whats specified. I think you need to allow extra properties?

File: backend/src/middlewares/rate-limit.ts [hunk 4]
```diff
@@ -498,6 +498,14 @@ export const apeKeysGenerate = rateLimit({
   handler: customHandler,
 });
 
+// Webhooks Routing
+export const webhooksPost = rateLimit({
+  windowMs: ONE_HOUR_MS,
+  max: 120 * REQUEST_MULTIPLIER,
```
Comment (Miodec) [inline backend/src/middlewares/rate-limit.ts:504]: 10 max

File: backend/src/api/controllers/webhook.ts (hunk 2 above)
Comment (Ferotiq) [reply backend/src/api/controllers/webhook.ts:10]: this isn't regenerating, it's comparing the hashes, afaict github auto hashes the secret. I could be wrong tho

Comment (Ferotiq) [reply backend/src/api/controllers/webhook.ts:10]: ig the hash could just be stored in the env lol

Comment (Miodec) [reply backend/src/api/controllers/webhook.ts:10]: yeah nevermind i just read the github docs, keep it the way it is

File: backend/src/api/routes/webhooks.ts (hunk 1 above)
Comment (Ferotiq) [reply backend/src/api/routes/webhooks.ts:15]: I used `.unknown()` which allows unknown properties.

**Pull request closed without merging**